api_tests/files/views/test_file_detail.py

```diff
@@ -478,104 +496,114 @@ def test_listing(self):
-        assert_equal(res.status_code, 200)
-        assert_equal(len(res.json['data']['attributes']['tags']), 1)
-        assert_equal(res.json['data']['attributes']['tags'][0], 'goofier')
-
-    def test_tags_add_and_remove_properly(self):
-        self.app.put_json_api(self.url, self.payload, auth=self.user.auth)
-        self.payload['data']['attributes']['tags'] = []
-        res = self.app.put_json_api(self.url, self.payload, auth=self.user.auth)
-        assert_equal(res.status_code, 200)
-        assert_equal(len(res.json['data']['attributes']['tags']), 0)
-
-    def test_put_wo_tags_doesnt_remove_tags(self):
-        self.app.put_json_api(self.url, self.payload, auth=self.user.auth)
-        self.payload['data']['attributes'] = {'checkout': None}
-        res = self.app.put_json_api(self.url, self.payload, auth=self.user.auth)
-        assert_equal(res.status_code, 200)
+        payload['data']['attributes']['tags'] = ['goofier']
+        res = app.put_json_api(url, payload, auth=user.auth)
+        assert res.status_code == 200
+        assert len(res.json['data']['attributes']['tags']) == 1
+        assert res.json['data']['attributes']['tags'][0] == 'goofier'
+
+    def test_tags_add_and_remove_properly(self, app, user, url, payload):
+        app.put_json_api(url, payload, auth=user.auth)
+        payload['data']['attributes']['tags'] = []
+        res = app.put_json_api(url, payload, auth=user.auth)
+        assert res.status_code == 200
+        assert len(res.json['data']['attributes']['tags']) == 0
+
+    def test_put_wo_tags_doesnt_remove_tags(self, app, user, url, payload):
+        app.put_json_api(url, payload, auth=user.auth)
+        payload['data']['attributes'] = {'checkout': None}
+        res = app.put_json_api(url, payload, auth=user.auth)
+        assert res.status_code == 200
         # Ensure adding tag data is correct from the PUT response
-        assert_equal(len(res.json['data']['attributes']['tags']), 1)
-        assert_equal(res.json['data']['attributes']['tags'][0], 'goofy')
-
-    def test_add_tag_adds_log(self):
-        count = self.node.logs.count()
-        self.app.put_json_api(self.url, self.payload, auth=self.user.auth)
-        assert_equal(self.node.logs.count(), count + 1)
-        assert_equal(NodeLog.FILE_TAG_ADDED, self.node.logs.latest().action)
-
-    def test_remove_tag_adds_log(self):
-        self.app.put_json_api(self.url, self.payload, auth=self.user.auth)
-        self.payload['data']['attributes']['tags'] = []
-        count = self.node.logs.count()
-        self.app.put_json_api(self.url, self.payload, auth=self.user.auth)
-        assert_equal(self.node.logs.count(), count + 1)
-        assert_equal(NodeLog.FILE_TAG_REMOVED, self.node.logs.latest().action)
+        assert len(res.json['data']['attributes']['tags']) == 1
+        assert res.json['data']['attributes']['tags'][0] == 'goofy'
+
+    def test_add_and_remove_tag_adds_log(self, app, user, url, payload, node):
```

Comment:
Looks like the same thing here, there appears to be some mutation that isn't overlapping, but grouping them might not be a good decision. @alexschiller